apps/communications/contacts/js/views/selector.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+/* global Contacts, contacts, utils */
+
+Contacts.Selector = (function(){
+  var selectorHeader = document.getElementById('selector-view-header');
+  var saveButton = document.getElementById('save-button');
+  var container = document.getElementById('selector-container');
+  var template = document.getElementById('contactToImportTemplate');
+
+  saveButton.addEventListener('click', importAll);
+  selectorHeader.addEventListener('action', cancelHandler);
+  container.addEventListener('click', clickHandler);
+
+  var contactsToImport = [];
+
+  function cancelHandler() {
+    selectorHeader.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    container.removeEventListener('click', clickHandler);
+    Contacts.cancel();
+  }
+
+  function clickHandler(event) {
+    // FIXME Seems to be a conflict when calling an activity from an activity.
+    /*var id = event.target.dataset.tempId;
+    ActivityHandler.mozContactParam = contactsToImport[id];
+    var activity = {
+      source: {
+        name: 'new',
+        data: {
+          type: ['text/vcard'],
+          allowSave: true,
+          params: {
+            mozContactParam: true
+          }
+        }
+      }
+    };
+    ActivityHandler.handle(activity);*/
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+
+    contactsToImport.forEach((contact, index) => {
+      contacts.Matcher.match(contact, 'passive', {
+        onmatch: matches => {
+          var callbacks = {
+            success: function(mergedContact) {
+              numDupsMerged++;
+              _doContinue(index);
+            },
+            error: _doContinue.bind(this, index)
```

Comment:
I will change it to `() => _doContinue(index)` which seems less obscure. I just used it to save an anonymous function declaration.